src/redisearch_rs/buffer/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! The `Buffer` API Rust wrappers.
+//!
+//! This crate provides Rust wrappers for the `Buffer`, `BufferReader` and `BufferWriter` structs
+//! from `buffer.h`.
+
+use std::{
+    ptr::{NonNull, copy_nonoverlapping},
+    slice,
+};
+
+/// Redefines the `Buffer` struct from `buffer.h`
+#[repr(C)]
+pub struct Buffer {
```

Comment:
I'm assuming that we want the user to guarantee that all elements up to `len` are initialized whenever we get a `Buffer` on our hands.

Reply:
sure. I assume you mean that we we should add a `Safety` section in the doc comment with this and other invariants listed?

Reply:
Yup!